text/0000-inline-asm.md

```diff
@@ -0,0 +1,869 @@
+- Feature Name: `asm`
+- Start Date: (fill me in with today's date, YYYY-MM-DD)
+- RFC PR: [rust-lang/rfcs#0000](https://github.com/rust-lang/rfcs/pull/0000)
+- Rust Issue: [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+
+# Summary
+[summary]: #summary
+
+This RFC specifies a new syntax for inline assembly which is suitable for eventual stabilization.
+
+The initial implementation of this feature will focus on the ARM, x86 and RISC-V architectures. Support for more architectures will be added based on user demand.
+
+The existing `asm!` macro will be renamed to `llvm_asm!` to provide an easy way to maintain backwards-compatibility with existing code using inline asm. However `llvm_asm!` is not intended to ever be stabilized.
```

Comment:
Will there be some transition period where both `asm!` and `llvm_asm!` work like the current `asm!` and some lint warns against using `asm!`? I know `asm!` is unstable and all, but quite a few crates use the current `asm!` and changing the semantics of it without some heads-up could be a bit frustrating? Doesn't even need to be very long, 6 weeks or so should probably do.

Reply:
Yes that is the intent, see #2843 for details on `llvm_asm!`.

Reply:
Ah perfect! A link that that RFC could be included in this RFC then :)